diff -N -U 3 -r firefox-orig/defaults/pref/firefox.js firefox/defaults/pref/firefox.js
--- firefox-orig/defaults/pref/firefox.js	2010-01-16 00:41:43.000000000 +0200
+++ firefox/defaults/pref/firefox.js	2010-01-23 11:58:18.000000000 +0200
@@ -782,3 +782,6 @@
 pref("toolbar.customization.usesheet", false);
 //@line 911 "/builds/slave/linux_build/build/browser/app/profile/firefox.js"
 
+// use Thunderbird as the default external email app
+ pref("network.protocol-handler.app.mailto", "/usr/bin/thunderbird");
+
diff -N -U 3 -r firefox-orig/defaults/profile/mimeTypes.rdf firefox/defaults/profile/mimeTypes.rdf
--- firefox-orig/defaults/profile/mimeTypes.rdf	2010-01-16 00:57:24.000000000 +0200
+++ firefox/defaults/profile/mimeTypes.rdf	2010-01-23 11:58:18.000000000 +0200
@@ -1,13 +1,28 @@
-<?xml version="1.0"?>  
-
-<RDF xmlns="http://www.w3.org/1999/02/22-rdf-syntax-ns#" 
-     xmlns:NC="http://home.netscape.com/NC-rdf#"
-     xmlns:RDF="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
-
-  <Description about="urn:mimetypes"> 
-    <NC:MIME-types> 
-      <Seq about="urn:mimetypes:root"> 
-      </Seq> 
-    </NC:MIME-types> 
-  </Description> 
-</RDF>
+<?xml version="1.0"?>
+<RDF:RDF xmlns:NC="http://home.netscape.com/NC-rdf#"
+         xmlns:RDF="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
+  <RDF:Seq RDF:about="urn:mimetypes:root">
+  </RDF:Seq>
+  <RDF:Description RDF:about="urn:mimetypes">
+    <NC:MIME-types RDF:resource="urn:mimetypes:root"/>
+  </RDF:Description>
+  <RDF:Description RDF:about="urn:scheme:mailto"
+                   NC:value="mailto">
+    <NC:handlerProp RDF:resource="urn:scheme:handler:mailto"/>
+  </RDF:Description>
+  <RDF:Description RDF:about="urn:handler:web:http://compose.mail.yahoo.com/?To=%s"
+                   NC:prettyName="Yahoo! Mail"
+                   NC:uriTemplate="http://compose.mail.yahoo.com/?To=%s" />
+  <RDF:Description RDF:about="urn:handler:web:https://mail.google.com/mail/?extsrc=mailto&amp;url=%s"
+                   NC:prettyName="GMail"
+                   NC:uriTemplate="https://mail.google.com/mail/?extsrc=mailto&amp;url=%s" />
+  <RDF:Description RDF:about="urn:scheme:handler:mailto"
+                   NC:alwaysAsk="true">
+    <NC:possibleApplication RDF:resource="urn:handler:web:http://compose.mail.yahoo.com/?To=%s"/>
+    <NC:possibleApplication RDF:resource="urn:handler:web:https://mail.google.com/mail/?extsrc=mailto&amp;url=%s"/>
+    <NC:possibleApplication RDF:resource="urn:handler:local:/usr/bin/thunderbird"/>
+  </RDF:Description>
+  <RDF:Description RDF:about="urn:handler:local:/usr/bin/thunderbird"
+                   NC:prettyName="thunderbird"
+                   NC:path="/usr/bin/thunderbird" />
+</RDF:RDF>
diff -N -U 3 -r firefox-orig/firefox firefox/firefox
--- firefox-orig/firefox	2010-01-16 00:41:33.000000000 +0200
+++ firefox/firefox	2010-01-23 11:58:18.000000000 +0200
@@ -50,7 +50,15 @@
 #uncomment for debugging
 #set -x
 
-moz_libdir=/usr/local/lib/firefox-3.6
+moz_libdir=/usr/lib/firefox-3.6
+
+# Include /usr/lib/mozilla/plugins in the plugin path:
+if [ "$MOZ_PLUGIN_PATH" ] ; then
+  MOZ_PLUGIN_PATH=$MOZ_PLUGIN_PATH:${moz_libdir}/plugins:/usr/lib/mozilla/plugins
+else
+  MOZ_PLUGIN_PATH=${moz_libdir}/plugins:/usr/lib/mozilla/plugins
+fi
+export MOZ_PLUGIN_PATH
 
 # Use run-mozilla.sh in the current dir if it exists
 # If not, then start resolving symlinks until we find run-mozilla.sh
